dot/core/messages.go

```diff
@@ -70,32 +74,34 @@ func (s *Service) HandleTransactionMessage(peerID peer.ID, msg *network.Transact
 		return false, err
 	}
 
-	allTxnsAreValid := true
+	allTxsAreValid := true
 	for _, tx := range txs {
-		txnIsValid := true
+		isValidTxn := true
 		validity, err := s.validateTransaction(head, rt, tx)
 		if err != nil {
-			txnIsValid = false
-			allTxnsAreValid = false
-			switch err := err.(type) {
+			switch err.(type) {
 			case runtime.InvalidTransaction:
+				isValidTxn = false
+				allTxsAreValid = false
```

Comment:
I would suggest using `continue` right after reporting the peer 
```suggestion
				s.net.ReportPeer(peerset.ReputationChange{
					Value:  peerset.BadTransactionValue,
					Reason: peerset.BadTransactionReason,
				}, peerID)
				continue
```